solr/solrj/src/test/org/apache/solr/common/util/TestJavaBinCodec.java

```diff
@@ -287,6 +289,45 @@ public void testAllTypes() throws IOException {
     );
   }
 
+  @Test
+  public void testReadMapEntry() throws IOException {
+    Map.Entry<Object, Object> extryFromTextDoc1 = getMapFromJavaBinCodec(SOLRJ_DOCS_1);
+    Map.Entry<Object, Object> extryFromTextDoc1_clone = getMapFromJavaBinCodec(SOLRJ_DOCS_1);
+
+    Map.Entry<Object, Object> extryFromTextDoc2 = getMapFromJavaBinCodec(SOLRJ_DOCS_2);
+    Map.Entry<Object, Object> extryFromTextDoc2_clone = getMapFromJavaBinCodec(SOLRJ_DOCS_2);
+
+    // exactly same document read twice should have same content
+    assert (extryFromTextDoc1.equals(extryFromTextDoc1_clone));
+    // doc1 and doc2 are 2 text files with different content on line 1
+    assert (!extryFromTextDoc2.equals(extryFromTextDoc1));
+    // exactly same document read twice should have same content
+    assert (extryFromTextDoc2.equals(extryFromTextDoc2_clone));
+
+    // now lets look at binary files
```

Comment:
suggestion, maybe take this to a different method?

Reply:
sure